src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -982,16 +985,59 @@ def customizePerJob(self, job):
 
         # Performance estimates
         if job.get('estimatedJobTime'):
-            jdl.append('+MaxWallTimeMins = %d\n' % int(job['estimatedJobTime'] / 60.0))
+            estimatedMins = int(job['estimatedJobTime'] / 60.0)
+            origCores = int(job.get('numberOfCores', 1))
+            estimatedMinsSingleCore = estimatedMins*origCores
+            # For now, assume a 15 minute job startup overhead -- condor will round this up further
+            jdl.append('+EstimatedSingleCoreMins = %d\n' % estimatedMinsSingleCore)
+            jdl.append('+OriginalMaxWallTimeMins = %d\n' % estimatedMins)
+            jdl.append('+MaxWallTimeMins = WMCore_ResizeJob ? (EstimatedSingleCoreMins/RequestCpus + 15) : OriginalMaxWallTimeMins\n')
+            
         if job.get('estimatedMemoryUsage'):
-            jdl.append('request_memory = %d\n' % int(job['estimatedMemoryUsage']))
+            baseMem = int(job['estimatedMemoryUsage'])
+            jdl.append('+OriginalMemory = %d\n' % baseMem)
```

Comment:
And Farrukh just mentioned Merge jobs don't carry OriginalMemory classAd, now I see why.

Perhaps we should always set these classAds to a reasonable number when estimated numbers are not found, like 1250 MB